lldb/source/Plugins/ABI/LoongArch/ABISysV_loongarch.cpp

```diff
@@ -0,0 +1,664 @@
+                            uint8_t size_in_bytes, bool is_signed) {
+  switch (size_in_bytes) {
+  default:
+    return false;
+
+  case sizeof(uint64_t):
+    SetInteger<uint64_t>(scalar, raw_value, is_signed);
+    break;
+
+  case sizeof(uint32_t):
+    SetInteger<uint32_t>(scalar, raw_value, is_signed);
+    break;
+
+  case sizeof(uint16_t):
+    SetInteger<uint16_t>(scalar, raw_value, is_signed);
+    break;
+
+  case sizeof(uint8_t):
+    SetInteger<uint8_t>(scalar, raw_value, is_signed);
+    break;
+  }
+
+  return true;
+}
+
+static bool SetSizedFloat(Scalar &scalar, uint64_t raw_value,
+                          uint8_t size_in_bytes) {
+  switch (size_in_bytes) {
+  default:
+    return false;
+
+  case sizeof(uint64_t):
+    scalar = *reinterpret_cast<double *>(&raw_value);
+    break;
+
+  case sizeof(uint32_t):
+    scalar = *reinterpret_cast<float *>(&raw_value);
+    break;
+  }
+
+  return true;
+}
+
+static ValueObjectSP GetValObjFromIntRegs(Thread &thread,
+                                          const RegisterContextSP &reg_ctx,
+                                          llvm::Triple::ArchType machine,
+                                          uint32_t type_flags,
+                                          uint32_t byte_size) {
+  Value value;
+  ValueObjectSP return_valobj_sp;
+  auto reg_info_a0 =
+      reg_ctx->GetRegisterInfo(eRegisterKindGeneric, LLDB_REGNUM_GENERIC_ARG1);
+  auto reg_info_a1 =
+      reg_ctx->GetRegisterInfo(eRegisterKindGeneric, LLDB_REGNUM_GENERIC_ARG2);
+  uint64_t raw_value;
+
+  switch (byte_size) {
+  case sizeof(uint32_t):
+    // Read a0 to get the arg
+    raw_value = reg_ctx->ReadRegisterAsUnsigned(reg_info_a0, 0) & UINT32_MAX;
+    break;
+  case sizeof(uint64_t):
+    raw_value = reg_ctx->ReadRegisterAsUnsigned(reg_info_a0, 0);
+    break;
+  case 16: {
+    // Create the ValueObjectSP here and return
+    std::unique_ptr<DataBufferHeap> heap_data_up(
+        new DataBufferHeap(byte_size, 0));
+    const ByteOrder byte_order = thread.GetProcess()->GetByteOrder();
+    RegisterValue reg_value_a0, reg_value_a1;
+    if (reg_ctx->ReadRegister(reg_info_a0, reg_value_a0) &&
+        reg_ctx->ReadRegister(reg_info_a1, reg_value_a1)) {
+      Status error;
+      if (reg_value_a0.GetAsMemoryData(*reg_info_a0,
+                                       heap_data_up->GetBytes() + 0, 8,
+                                       byte_order, error) &&
+          reg_value_a1.GetAsMemoryData(*reg_info_a1,
+                                       heap_data_up->GetBytes() + 8, 8,
+                                       byte_order, error)) {
+        value.SetBytes(heap_data_up.release(), byte_size);
+        return ValueObjectConstResult::Create(
+            thread.GetStackFrameAtIndex(0).get(), value, ConstString(""));
+      }
+    }
+    break;
+  }
+  default:
+    return return_valobj_sp;
+  }
+
+  if (type_flags & eTypeIsInteger) {
+    const bool is_signed = (type_flags & eTypeIsSigned) != 0;
```

Comment:
Just put this in the call to SetSizedInteger rather than making another local variable.

Reply:
Thanks, done.